### DISCUSSION TITLE
`channel` field in `Provider` of type `slack` is redundant?

### DISCUSSION THREAD

**Discussion by or-shachar** [fluxcd/flux2#1700] [Q&A]
When configuring Slack provider I'm required to provide "webhook URL" (via direct property or via secret ref).

The Provider spec also allow me to define `channel`.

IIUC - simple Slack webhook URL can only be generated with permissions to post to specific channel, so I can't really configure webhook URL and different channel name.

I wonder if anyone succeeded to configure multiple Slack Providers on different channels using the `channel` field, but using the same webhook URL.

If it's impossible to use same webhook url on different channels - maybe it's best to clarify this in the example (https://fluxcd.io/docs/components/notification/provider/#notifications).

**Comment by seh** [comment]
I brought this up in [the "flux" channel of the CNCF Slack workspace on 28 May 2021](https://cloud-native.slack.com/archives/CLAJ40HV3/p1622224353095900), but received no reply. It's clear that Flux demands that the field be populated, but it doesn't look like Slack needs it, or pays any attention to it.

**Reply by stefanprodan** [reply]
The Slack provider is also used for other platforms that are mimicking the Slack API, for those platforms like Mattermost the channel is required.

**Reply by or-shachar** [reply]
1. Maybe it's worth documenting this somewhere to avoid confusion.
2. Do we have thoughts about supporting Slack [postMessage API](https://api.slack.com/methods/chat.postMessage) and using a token from k8s secret?  IIUC - this API supports different channels, and if the token was given the right scope - can allow Flux to post a message to any channel that is stated in the `channel` field.

**Reply by stefanprodan** [reply]
We can't change the current implementation because it will break Flux when used with Slack clones. I'm for adding a new alert provider e.g. `SlackBot` that would use the postMessage API. PR welcome!
